core/templates/services/platform-feature.service.ts

```diff
@@ -0,0 +1,285 @@
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface FeatureFlagsCacheItem {
+  timestamp: number;
+  sessionId: string;
+  featureStatusSummary: FeatureStatusSummary;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PlatformFeatureService {
+  private static SESSION_STORAGE_KEY = 'SAVED_FEATURE_FLAGS';
+  private static SESSION_STORAGE_CACHE_TTL = 12 * 3600 * 1000; // 12 hours
+
+  private static COOKIE_NAME_FOR_SESSION_ID = 'SACSID';
+  private static COOKIE_NAME_FOR_SESSION_ID_IN_DEV = 'dev_appserver_login';
+
+  // TODO(#9154): Remove static when migration is complete.
+  static featureStatusSummary: FeatureStatusSummary = null;
+
+  // TODO(#9154): Remove static when migration is complete.
+  static _initializedWithError = false;
+
+  // TODO(#9154): Remove static when migration is complete.
+  static initializationPromise: Promise<void> = null;
+
+  constructor(
+      private clientContextObjectFactory: ClientContextObjectFactory,
+      private platformFeatureBackendApiService:
+        PlatformFeatureBackendApiService,
+      private featureStatusSummaryObjectFactory:
+        FeatureStatusSummaryObjectFactory,
+      private i18nLanguageCodeService: I18nLanguageCodeService,
+      private windowRef: WindowRef) {
+    this.initialize();
+  }
+
+  /**
+   * Inializes the PlatformFeatureService, it's guaranteed that it's only
+   * inialized once.
+   *
+   * @returns {Promise} - A promise that is resolved when the initialization
+   * is done.
+   */
+  async initialize(): Promise<void> {
+    if (!PlatformFeatureService.initializationPromise) {
+      PlatformFeatureService.initializationPromise = this._initialize();
+    }
+    return PlatformFeatureService.initializationPromise;
+  }
+
+  /**
+   * Returns the summary object of feature flags, which can be used to get the
+   * value of feature flags.
+   * Example:
+   *   platformFeatureService.featureSummary.DummyFeature.isEnabled === true.
+   *
+   * @returns {FeatureSummaryDict} - Summary object of feature flags.
+   * @throws {Error} - If this method is called before inialization.
+   */
+  get featureSummary(): FeatureSummaryDict {
+    if (PlatformFeatureService.featureStatusSummary) {
+      return PlatformFeatureService.featureStatusSummary.toSummaryDict();
+    } else {
+      throw new Error('The platform feature service has not been initialized.');
+    }
+  }
+
+  /**
+   * Checks if there's any error, e.g. request timeout, during initialization.
+   *
+   * @returns {boolean} - True if there is any error during initialization.
+   */
+  get initialzedWithError(): boolean {
+    return PlatformFeatureService._initializedWithError;
+  }
+
+  /**
+   * Detects the type of browser from it's userAgent.
+   *
+   * @returns {string} - The name of the browser that is being used.
+   */
+  detectBrowserType(): string {
```

Comment:
Do we not already have a service to do this? If not, can we make this a separate service instead?